### PR TITLE
Basic client code

### PR DESCRIPTION
**Pull request by tarampampam**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes

## Description

Code prototype.

### CODE REVIEW

File: src/Frameworks/Illuminate/ClientFactory.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            throw new \InvalidArgumentException("Connection [$connection_name] does not exists.");
+        }
+
+        $connection_settings = $this->config->get("{$this->config_root}.connections.{$connection_name}");
+
+        $settings = new ConnectionSettings(
+            (string) $connection_settings['entrypoint'],
+            (string) $connection_settings['login'],
+            (string) $connection_settings['password'],
+            (int) $connection_settings['timeout'],
```
Comment (jetexe) [inline src/Frameworks/Illuminate/ClientFactory.php]: Timeout and UA not required

File: src/Frameworks/Illuminate/config/rabbitmq-api-client.php [hunk 2]
```diff
@@ -4,12 +4,36 @@
...
+    |
+    | Here are each of the RabbitMQ API connections setup for your application.
+    |
+    */
+
+    'connections' => [
+
+        'default-connection' => [
+            'entrypoint' => env('RABBITMQ_API_ENTRYPOINT'),
+            'login'      => env('RABBITMQ_API_LOGIN'),
```
Comment (jetexe) [inline src/Frameworks/Illuminate/config/rabbitmq-api-client.php]: `guest` `guest`

File: src/QueueInfo.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        return $this->raw_data['vhost'] ?? null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toJson($options = 0): string
+    {
+        return Json::encode($this->raw_data);
```
Comment (jetexe) [inline src/QueueInfo.php]: Прокинуть `$options`

File: tests/Frameworks/Illuminate/LaravelServiceProviderTest.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public function testRegistrationComponentsInIoc()
+    {
+        $this->assertInstanceOf(ClientFactory::class, $this->app->make(ClientFactoryInterface::class));
+        $this->assertInstanceOf(ClientInterface::class, $default_client = $this->app->make(ClientInterface::class));
```
Comment (jetexe) [inline tests/Frameworks/Illuminate/LaravelServiceProviderTest.php]: rm `$default_client = `

File: tests/Frameworks/Illuminate/ClientFactoryTest.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+     * {@inheritdoc}
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->factory = new ClientFactory($this->app->make(ConfigRepository::class));
+        $this->root    = LaravelServiceProvider::getConfigRootKeyName();
+    }
+
```
Comment (jetexe) [inline tests/Frameworks/Illuminate/ClientFactoryTest.php:57]: test guzzle option passthrough

**Review by jetexe — CHANGES_REQUESTED** [review]

File: src/Frameworks/Illuminate/ClientFactory.php (hunk 1 above)
Comment (tarampampam) [reply src/Frameworks/Illuminate/ClientFactory.php]: ready

File: src/Frameworks/Illuminate/config/rabbitmq-api-client.php (hunk 2 above)
Comment (tarampampam) [reply src/Frameworks/Illuminate/config/rabbitmq-api-client.php]: ready

File: src/QueueInfo.php (hunk 3 above)
Comment (tarampampam) [reply src/QueueInfo.php]: ready

File: tests/Frameworks/Illuminate/LaravelServiceProviderTest.php (hunk 4 above)
Comment (tarampampam) [reply tests/Frameworks/Illuminate/LaravelServiceProviderTest.php]: ready

File: tests/Frameworks/Illuminate/ClientFactoryTest.php (hunk 5 above)
Comment (tarampampam) [reply tests/Frameworks/Illuminate/ClientFactoryTest.php:57]: ready

**Review by Reallife — APPROVED** [review]

**Pull request merged**